packages/sanity/src/core/form/inputs/ReferenceInput/ReferenceInput.tsx

```diff
@@ -149,7 +149,8 @@ export function ReferenceInput(props: ReferenceInputProps) {
     (inputValue$: Observable<string | null>) => {
       return inputValue$.pipe(
         filter(nonNullable),
-        distinctUntilChanged(),
+        // ensures that the filter function is always fired when the button is clicked
+        distinctUntilChanged((prev, cur) => prev !== '' && cur !== ''),
```

Comment:
Would it be just as good to remove the whole `distinctUntilChanged` here? Since it's a stream of values emitted from the text input I don't think we will ever get two following identical values (the input element will only emit onChange when it's `value` actually changes, right?), or am I missing something?
```suggestion
        // ensures that the filter function is always fired when the button is clicked
        distinctUntilChanged((prev, cur) => prev !== '' && cur !== ''),
```

Reply:
That was my first thought, but I wasn't sure if it's there for a reason, I am happy to remove it if it makes more sense

Reply:
let's remove it - it seems to work exactly the same without it ✂️